java/cuvs-java/src/main/java22/com/nvidia/cuvs/spi/JDKProvider.java

```diff
@@ -57,4 +59,9 @@ public CagraIndex.Builder newCagraIndexBuilder(CuVSResources cuVSResources) {
   public HnswIndex.Builder newHnswIndexBuilder(CuVSResources cuVSResources) {
     return HnswIndexImpl.newBuilder(Objects.requireNonNull(cuVSResources));
   }
+
+  @Override
+  public Dataset newDataset(int size, int dimensions) throws UnsupportedOperationException {
+	  return new DatasetImpl(size, dimensions);
```

Comment:
Nit:  The indentation seems to be off a little here.  More like 6 spaces than 2.
```suggestion
      return new DatasetImpl(size, dimensions);
```